src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lateralSide;
+    private double upperBase;
+    private double bottomBase;
+    private double height;
+    private double triangleLeg;
+
+    public IsoscelesTrapezoid() {
+    }
+
+    public IsoscelesTrapezoid(double upperBase, double bottomBase, double lateralSide) {
+        this.typeOfFigure = TypeOfFigure.ISOSCELES_TRAPEZOID;
+        this.area = calculateArea(upperBase, bottomBase, triangleLeg, height);
+        this.upperBase = upperBase;
+        this.bottomBase = bottomBase;
+        this.lateralSide = lateralSide;
+        this.triangleLeg = (bottomBase - upperBase) / 2;
+        this.height = getTrapezoidHigh(lateralSide, triangleLeg);
```

Comment:
You should not calculate the height in the constructor. It's better to calculate it in the `calculateArea` method or in a separate method that is called within `calculateArea`.